AnimationGroup.lua

```diff
@@ -60,28 +60,55 @@ function CreateFrame(...)
     return frame
 end
 
-
 local function OnUpdate(self, elapsed)
     if self.paused then
         return
     end
-
-    self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+    local reverse = self.group.reverse
+    local in_start_delay = self.startDelay and self.startDelayTime < self.startDelay
+    local in_end_delay = self.endDelay and self.endDelayTime < self.endDelay
+    local in_progress = not reverse and self.time < self.duration
+    local in_progress_revers = reverse and 0 < self.time
+    self.delaying = true
+
+    if not reverse and in_start_delay then
+        self.startDelayTime = self.startDelayTime + elapsed
+    elseif reverse and in_end_delay then
+        self.endDelayTime = self.endDelayTime + elapsed
+    elseif (in_progress) or (in_progress_revers) then
+        self.time = self.time + (self.group.reverse and -elapsed or elapsed)
+        self.delaying = false
+        self.progress = self.time / self.duration
+        self.progress = self.progress < 0 and 0 or self.progress
+        self.progress = self.progress > 1 and 1 or self.progress
+        self.smoothProgress = self.smoothing_func(self.progress).y
+    end
 
     if self.time > self.duration or (self.group.reverse and self.time < 0) then
```

Comment:
```suggestion
    local is_animation_complete = self.duration < self.time or (self.group.reverse and self.time < 0)
    if is_animation_complete and not self.delayed then
```

I suppose you don't need to add the `elapsed` twice (this block and the previous) - just leave the OnUpdate firing when the animation is delayed.